Missionframework/scripts/client/ui/ui_manager.sqf

```diff
@@ -66,124 +31,91 @@ waitUntil { synchro_done };
             KP_liberation_supplies = KP_liberation_supplies_global;
             KP_liberation_ammo = KP_liberation_ammo_global;
             KP_liberation_fuel = KP_liberation_fuel_global;
         } else {
-            _resource_area = toUpper ([_nearfob] call KPLIB_fnc_getFobName);
-            KP_liberation_supplies = ((_actual_fob select 0) select 1);
-            KP_liberation_ammo = ((_actual_fob select 0) select 2);
-            KP_liberation_fuel = ((_actual_fob select 0) select 3);
+            KP_liberation_supplies = _supplies;
+            KP_liberation_ammo = _ammo;
+            KP_liberation_fuel = _fuel;
         };
-        KP_liberation_air_vehicle_building_near = ((_actual_fob select 0) select 4);
-        KP_liberation_recycle_building_near = ((_actual_fob select 0) select 5);
+        // TODO this is used by build scripts, move to relevant places
+        KP_liberation_air_vehicle_building_near = _hasAir;
+        KP_liberation_recycle_building_near = _hasRecycling;
     } else {
-        _resources = false;
+        _showResources = false;
         KP_liberation_supplies = 0;
         KP_liberation_ammo = 0;
         KP_liberation_fuel = 0;
         KP_liberation_air_vehicle_building_near = false;
         KP_liberation_recycle_building_near = false;
     };
 
-    if ( _overlayshown) then {
+    if ( _overlayVisible) then {
 
-        ((uiNamespace getVariable 'GUI_OVERLAY') displayCtrl (266)) ctrlSetText format [ "%1", GRLIB_ui_notif ];
-        ((uiNamespace getVariable 'GUI_OVERLAY') displayCtrl (267)) ctrlSetText format [ "%1", GRLIB_ui_notif ];
+        (_overlay displayCtrl (266)) ctrlSetText format [ "%1", GRLIB_ui_notif ];
+        (_overlay displayCtrl (267)) ctrlSetText format [ "%1", GRLIB_ui_notif ];
 
         if ((markerPos "opfor_capture_marker") distance markers_reset > 100 ) then {
 
             private [ "_attacked_string" ];
             _attacked_string = [markerpos "opfor_capture_marker"] call KPLIB_fnc_getLocationName;
 
-            ((uiNamespace getVariable 'GUI_OVERLAY') displayCtrl (401)) ctrlShow true;
-            ((uiNamespace getVariable 'GUI_OVERLAY') displayCtrl (402)) ctrlSetText _attacked_string;
-            ((uiNamespace getVariable 'GUI_OVERLAY') displayCtrl (403)) ctrlSetText (markerText "opfor_capture_marker");
+            (_overlay displayCtrl (401)) ctrlShow true;
+            (_overlay displayCtrl (402)) ctrlSetText _attacked_string;
+            (_overlay displayCtrl (403)) ctrlSetText (markerText "opfor_capture_marker");
         } else {
-            ((uiNamespace getVariable 'GUI_OVERLAY') displayCtrl (401)) ctrlShow false;
-            ((uiNamespace getVariable 'GUI_OVERLAY') displayCtrl (402)) ctrlSetText "";
-            ((uiNamespace getVariable 'GUI_OVERLAY') displayCtrl (403)) ctrlSetText "";
+            (_overlay displayCtrl (401)) ctrlShow false;
+            (_overlay displayCtrl (402)) ctrlSetText "";
+            (_overlay displayCtrl (403)) ctrlSetText "";
```

Comment:
All these control numbers could be defined like you did it in the overlayUpdateResources function.